services/QuillLMS/spec/models/activity_session_spec.rb

```diff
@@ -955,4 +955,22 @@
       expect(activity_session.timespent).to eq(99)
     end
   end
+
+
+  describe "#teacher_activity_feed" do
+    it "should create a teacher_activity_feed item once on completed." do
+      activity_session = create(:activity_session, completed_at: nil)
+      teacher = activity_session.teachers.first
+
+      activity_session.update(completed_at: Time.now)
+      activity_session.save
+      activity_session.save
```

Comment:
Why do we have an update and then two saves here?

Reply:
> Why do we have an update and then two saves here?

So this was an attempt to verify that multiple saves will not trigger multiple adds to the Activity Feed. I'm going to re-write this to make that clearer.